services/thermal_mgr/thermal_mgr.c

```diff
@@ -42,18 +42,75 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 }
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
-  /* Send an event to the thermal manager queue */
+  if (thermalMgrQueueHandle == NULL)
+    return ERR_CODE_INVALID_STATE;
+
+  if (event == NULL)
+    return NULL_ERR;
+
+  if (xQueueSend(thermalMgrQueueHandle, event, 0) != pdTRUE)
+    return ERR_CODE_QUEUE_FULL;
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event;
+  event.type = EVENT_INTERRUPT;
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+
+  // Get config data from pvParameters
+  lm75bd_config_t config_data = *(lm75bd_config_t *) pvParameters;
+
   while (1) {
+      thermal_mgr_event_t eventFromQueue;
+      error_code_t errorCode;
+  
+      // You can receive the next item in the queue using xQueueReceive
+      // if receive from queue is successful
+    	if (xQueueReceive(thermalMgrQueueHandle, &eventFromQueue, 0) == pdTRUE) {
+        float temperature;
+		  // Handle event received from queue
+        if( eventFromQueue.type == THERMAL_MGR_EVENT_MEASURE_TEMP_CMD){
+
+           errorCode = readTempLM75BD(config_data.devAddr, &temperature);
+           if (errorCode != ERR_CODE_SUCCESS){
+            LOG_ERROR_CODE(errorCode);
+            break;
```

Comment:
just continue instead of breaking since you don't want to kill the task just because of a single error